Source/CarthageKit/Project.swift

```diff
@@ -1095,14 +1095,42 @@ private func platformForFramework(_ frameworkURL: URL) -> SignalProducer<Platfor
 				return .readFailed(frameworkURL, error)
 			}
 
-			guard let sdkName = bundle?.object(forInfoDictionaryKey: "DTSDKName") else {
-				return .failure(readFailed("the DTSDKName key in its plist file is missing"))
+			func sdkNameFromExecutable() -> String? {
+				guard let executableURL = bundle?.executableURL else {
+					return nil
+				}
+
+				let task = Task("/usr/bin/otool", arguments: ["-lv", executableURL.path])
+
+				let sdkName: String? = task.launch(standardInput: nil)
+					.ignoreTaskData()
+					.map { String(data: $0, encoding: .utf8) ?? "" }
+					.filter { !$0.isEmpty }
+					.flatMap(.merge) { (output: String) -> SignalProducer<String, NoError> in
+						output.linesProducer
+					}
+					.filter { $0.contains("LC_VERSION") }
+					.take(last: 1)
+					.flatMap(.merge) { lcVersionLine -> SignalProducer<String?, NoError> in
+						let sdkString = lcVersionLine.split(separator: "_")
+							.last
+							.flatMap(String.init)
+							.flatMap { $0.lowercased() }
+
+						return .init(value: sdkString)
+					}.skipNil()
+					.single()?
+					.value
+
+				return sdkName
 			}
 
-			if let sdkName = sdkName as? String {
-				return .success(sdkName)
+			if let sdkNameFromBundle = bundle?.object(forInfoDictionaryKey: "DTSDKName") as? String {
+				return .success(sdkNameFromBundle)
+			} else if let sdkNameFromExecutable = sdkNameFromExecutable() {
```

Comment:
Can you add a comment before the `if`/`else` that explains why we're checking both and in which order?

Reply:
Done